commit 1006ccc48c14d3715c569cca68dab54742207273
Author: Andreas Schwab <schwab@suse.de>
Date:   Fri Apr 18 23:43:19 2008 +0200

    Properly emulate AC_ENABLE_SHARED([no]) and others.
    
    * libltdl/m4/ltoptions.m4 (AC_ENABLE_SHARED): Check optional
    argument.
    (AC_ENABLE_STATIC): Likewise.
    (AC_ENABLE_FAST_INSTALL): Likewise.
    
    Signed-off-by: Ralf Wildenhues <Ralf.Wildenhues@gmx.de>

diff --git a/ChangeLog b/ChangeLog
index 268e369..e593406 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2008-04-18  Andreas Schwab  <schwab@suse.de>
+
+	* libltdl/m4/ltoptions.m4 (AC_ENABLE_SHARED): Check optional
+	argument.
+	(AC_ENABLE_STATIC): Likewise.
+	(AC_ENABLE_FAST_INSTALL): Likewise.
+
 2008-04-16  Eric Blake  <ebb9@byu.net>
 
 	Make ChangeLog merges easier.
diff --git a/libltdl/m4/ltoptions.m4 b/libltdl/m4/ltoptions.m4
index a1c3aba..339d6f5 100644
--- a/libltdl/m4/ltoptions.m4
+++ b/libltdl/m4/ltoptions.m4
@@ -192,7 +192,7 @@ LT_OPTION_DEFINE([LT_INIT], [disable-shared], [_LT_ENABLE_SHARED([no])])
 
 # Old names:
 AC_DEFUN([AC_ENABLE_SHARED],
-[_LT_SET_OPTION([LT_INIT], [shared])
+[_LT_SET_OPTION([LT_INIT], m4_if([$1], [no], [disable-])[shared])
 ])
 
 AC_DEFUN([AC_DISABLE_SHARED],
@@ -246,7 +246,7 @@ LT_OPTION_DEFINE([LT_INIT], [disable-static], [_LT_ENABLE_STATIC([no])])
 
 # Old names:
 AC_DEFUN([AC_ENABLE_STATIC],
-[_LT_SET_OPTION([LT_INIT], [static])
+[_LT_SET_OPTION([LT_INIT], m4_if([$1], [no], [disable-])[static])
 ])
 
 AC_DEFUN([AC_DISABLE_STATIC],
@@ -300,7 +300,7 @@ LT_OPTION_DEFINE([LT_INIT], [disable-fast-install], [_LT_ENABLE_FAST_INSTALL([no
 
 # Old names:
 AU_DEFUN([AC_ENABLE_FAST_INSTALL],
-[_LT_SET_OPTION([LT_INIT], [fast-install])
+[_LT_SET_OPTION([LT_INIT], m4_if([$1], [no], [disable-])[fast-install])
 AC_DIAGNOSE([obsolete],
 [$0: Remove this warning and the call to _LT_SET_OPTION when you put
 the `fast-install' option into LT_INIT's first parameter.])
